crates/aptos/src/node/local_testnet/mod.rs

```diff
@@ -288,21 +315,47 @@ impl CliCommand<()> for RunLocalTestnet {
 
         eprintln!("\nSetup is complete, you can now use the local testnet!");
 
+        // Create a task that listens for ctrl-c. We want to intercept it so we can run
+        // the shutdown steps before properly exiting. This is of course best effort,
+        // see `ShutdownStep` for more info. In particular, to speak to how "best effort"
+        // this really is, to make sure ctrl-c happens more or less instantly, we only
+        // register this handler after all the services have started.
+        tasks.push(tokio::spawn(async move {
+            tokio::signal::ctrl_c()
+                .await
+                .expect("Failed to register ctrl-c hook");
+        }));
```

Comment:
Just making sure - So this register the ctrl-c hook and stops here, until hook is triggered then we continue forward to execute shutdown_steps? If I just close the terminal, I wonder if that count as `ctrl-c`?

Reply:
It doesn't stop exactly at this point, it stops at the select_all afterwards.

If someone closes the terminal then the CLI will receive a SIGHUP. We don't handle that right now so no cleanup will run. We could, but I'm choosing to only use things that are cross platform for now. Ctrl+C is meaningful on both Unix and Windows but SIGHUP is not.

There are other cases where there's nothing we can do to run the shutdown steps, like if the CLI gets SIGKILLed. So these shutdown steps are just best effort. It's not a huge deal if they don't run, the CLI will handle it on next run if the shutdown steps didn't happen.

I'll include something in the docs explaining how to clean up if the CLI exited uncleanly.